app/services/search/postgres/comment.rb

```diff
@@ -0,0 +1,118 @@
+module Search
+  module Postgres
+    class Comment
+      ATTRIBUTES = [
+        "COALESCE(articles.published, false) AS commentable_published",
+        "COALESCE(articles.title, '') AS commentable_title",
+        "comments.body_markdown",
+        "comments.commentable_id",
+        "comments.commentable_type",
+        "comments.created_at",
+        "comments.id AS id",
+        "comments.public_reactions_count",
+        "comments.score",
+        "comments.user_id",
+      ].freeze
+      private_constant :ATTRIBUTES
+
+      USER_ATTRIBUTES = %i[
+        id
+        name
+        profile_image
+        username
+      ].freeze
+      private_constant :USER_ATTRIBUTES
+
+      ARTICLE_COMMENTABLE_QUERY = <<-SQL.freeze
+        LEFT JOIN articles
+          ON comments.commentable_id = articles.id
+          AND comments.commentable_type = 'Article'
+      SQL
+      private_constant :ARTICLE_COMMENTABLE_QUERY
+
+      DEFAULT_PER_PAGE = 60
+      private_constant :DEFAULT_PER_PAGE
+
+      DEFAULT_SORT_BY = "comments.score".freeze
+      private_constant :DEFAULT_SORT_BY
+
+      DEFAULT_SORT_DIRECTION = :desc
+      private_constant :DEFAULT_SORT_DIRECTION
+
+      MAX_PER_PAGE = 120 # to avoid querying too many items, we set a maximum amount for a page
+      private_constant :MAX_PER_PAGE
+
+      # We filter comments for those that are:
+      # 1. On Articles
+      # 2. Not deleted
+      # 3. Not hidden by commentable user (i.e. an Article author didn't hide the comment)
+      # 4. Are attached to published articles
+      QUERY_FILTER = <<-SQL.freeze
+        comments.commentable_type = 'Article' AND
+        comments.deleted = false AND
+        comments.hidden_by_commentable_user = false AND
+        articles.published = true
+      SQL
+      private_constant :QUERY_FILTER
+
+      def self.search_documents(
+        page: 0,
+        per_page: DEFAULT_PER_PAGE,
+        sort_by: DEFAULT_SORT_BY,
+        sort_direction: DEFAULT_SORT_DIRECTION,
+        term: nil
+      )
+        # NOTE: [@rhymes/atsmith813] we should eventually update the frontend
+        # to start from page 1
+        sort_by ||= DEFAULT_SORT_BY
+        # The UI and serializer rename created_at (the actual DB column name) to
+        # published_at
+        sort_by = "comments.created_at" if sort_by == "published_at"
+
+        sort_direction ||= DEFAULT_SORT_DIRECTION
```

Comment:
Maybe I should move this to the controller since we'll need to do this for all the object types on the search page 🤔